ociclient/cache/cache.go

```diff
@@ -176,9 +176,11 @@ func (lc *layeredCache) get(dgst string, desc ocispecv1.Descriptor) (os.FileInfo
 		}
 		return nil, nil, err
 	}
-	// do simple blob verification
-	// todo: use digest to check if blob is valid
-	if info.Size() != desc.Size {
+	verified, err := verifyBlob(lc.baseFs.FileSystem, info, dgst, desc)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to verify blob: %w", err)
```

Comment:
Could there be an endless loop if blob in cache is damaged?

Reply:
depends on the user of the cache.
If the user ignores the error and defaults to fetch from the remote repo then we will not have a endlessloop.

In our implementation I think we have that behavior that we log the error but always default to the remote repository